### DISCUSSION TITLE
Django 4.0 and django-select2: redis changes?

### DISCUSSION THREAD

**Discussion by opbod** [codingjoe/django-select2#112] [Q&A]
Hi there,

I'm running into an issue with my Django 3.2 + gunicorn setup where django-select2 items through a `ModelSelect2Widget` aren't available at all times. 
When e.g. "apple" is provided as input, most of the times I get `The results could not be loaded` while typing.
The issue does not occur on my development install.
With a normal `Select2Widget`, the issue doesn't occur either.

When trying to debug, I see https://mysite/select2/fields/auto.json?term=appl&field_id=somefieldid is reached. This URL shows proper output (`{"results": [{"text": "Apple test", "id": 4}, {"text": "Apple test2", "id": 5}, <...>` in about 1 out of 10 tries. In about 9/10 cases I get a 404.

When looking at the [installation documentation](https://django-select2.readthedocs.io/en/7.4.2/), I see a persistent cache backend is a requirement for django-select2. Since I'm using multiple workers on my gunicorn config, I'm assuming the above problem is the exact reason for this requirement. Could this be confirmed? As I'm not familiar with Redis and/or do not have a caching need so far.

If so: Django 4.0 now comes with [native support of the Redis backend](https://docs.djangoproject.com/en/4.0/releases/4.0/#redis-cache-backend) through `django.core.cache.backends.redis.RedisCache`. 
Is the recommendation to use the native django support or to use `django_redis`'?

If native, is the below the `BACKEND` the only change that is required?
```
CACHES = {
    # … default cache config and others
    "select2": {
        "BACKEND": "django.core.cache.backends.redis.RedisCache",
        "LOCATION": "redis://127.0.0.1:6379/2",
        "OPTIONS": {
            "CLIENT_CLASS": "django_redis.client.DefaultClient",
        }
    }
}

# Tell select2 which cache configuration to use:
SELECT2_CACHE_BACKEND = "select2"
```

**Comment by codingjoe** [comment]
Did you check if your cache is working? You can do that from the shell, just import cache and do a simple set get.

**Reply by opbod** [reply]
Well, I'm not using a cache at the moment (AFAIK). No need for caching so far.

But since I'm pretty fond of using django-select2: is the randomness that I'm seeing related to the fact that I'm not using redis?
And is I need to use redis, do I go with the native one from Django 4.0 onwards (`django.core.cache.backends.redis.RedisCache`) or is the recommendation to use `django_redis`?

**Reply by codingjoe** [reply]
It doesn't matter, also memcached would work. Given that Django has its own Redis backend now, this would always be my go-to cache store.
